frontend/src/metabase-types/api/settings.ts

```diff
@@ -181,12 +188,17 @@ export type PasswordComplexity = {
 
 export type SessionCookieSameSite = "lax" | "strict" | "none";
 
-export interface SettingDefinition {
-  key: string;
+export interface SettingDefinition<
+  Key extends EnterpriseSettingKey = EnterpriseSettingKey,
+> {
+  key: Key;
   env_name?: string;
   is_env_setting: boolean;
-  value?: unknown;
-  default?: unknown;
+  value: EnterpriseSettings[Key] | null;
+  default?: EnterpriseSettings[Key];
+  display_name?: string;
+  description?: string | ReactNode;
```

Comment:
(I think ReactNode contains string, but it's good to communicate that we allow strings explicitly 👍🏻)